jdaviz/utils.py

```diff
@@ -493,8 +493,7 @@ def download_uri_to_path(possible_uri, cache=None, local_path=os.curdir, timeout
 
         if local_path is None:
             # if not specified, this is the default location:
-            local_path = os.path.join(os.getcwd(), parsed_uri.path.split(os.path.sep)[-1])
-
+            local_path = os.path.join(os.getcwd(), parsed_uri.path.split('/')[-1])
```

Comment:
Why was this not caught in CI? Is this not tested on Windows?

Reply:
I _might_ see what went wrong now, thanks @javerbukh. @pllim – it looks like urllib's `parse` method is for URLs and always uses `/`. This is a block in the code where the `parsed_uri.path` is parsed as a URL, which shouldn't be OS specific. Does that make sense?

Reply:
So, sounds like a new test needs to be added so this case is covered. And this new test should fail on `main` but pass with this patch. Correct?

Reply:
Do we have a test already of loading data with a windows type path @pllim ? If not, do you have an example of setting up a test like that?

Reply:
Given that most of the devs work off OSX, probably not. Here's an idea:

1. Find a small test that already use a data file (preferable not remote).
2. When you detect the test is running on Windows, force forward slash, maybe like this: `str(p.absolute()).replace("/", "\\")` (pathlib has `as_posix()` but it does not use forward slash when I use git bash shell)
3. See if this fails on `main`. You want it to fail on `main`.
4. See if your patch fixes it.

Reply:
+1 to @pllim's plan!

Reply:
I tried that and it does now work, feel free to test on your own windows machines with `pytest -s jdaviz/tests/test_utils.py --remote-data` .

Reply:
Jira ticket for adding remote data CI tests on Windows OS https://jira.stsci.edu/browse/JDAT-4596

Reply:
Should we just slap remote data on Windows job as part of this PR and take it back out when that other ticket is done?

Reply:
Would look something like this: https://github.com/spacetelescope/jdaviz/pull/2924

Reply:
Added that to the CI and it looks like it is fixed in this PR. Thank you for the example!